### PR TITLE
UX: New menu

### PR DESCRIPTION
**Pull request by jordanvidrine**
In this PR

- exported all code that changes the look & feel to a `@service`
- state is now tracked across topics using that same `@service`
- simplified the menu
- simplified font options

https://github.com/discourse/reader-mode/assets/30537603/cbff7598-d66d-4dd4-ae6d-7e819bc72fab

### CODE REVIEW

File: javascripts/discourse/components/reader-mode-options.gjs [hunk 1]
```diff
@@ -3,134 +3,147 @@ import { tracked } from "@glimmer/tracking";
...
       @identifier="reader-mode-options"
       @triggers={{array "click"}}
       @placementStrategy="fixed"
       @class="reader-mode-options"
       @inline={{true}}
+      {{didInsert this.readerMode.selectFontSize}}
+      {{didInsert this.readerMode.selectFont}}
+      {{didInsert this.readerMode.selectOffset}}
+      {{didInsert this.readerMode.setupWidth}}
+      {{didInsert this.readerMode.setupColors}}
```
Comment (janzenisaac) [inline javascripts/discourse/components/reader-mode-options.gjs:35]: Not necessary, but we could move these to a single function for a single `didInsert`.

Eg.

```suggestion
      {{didInsert this.setupReaderMode}}
```
```js
setupReaderMode(){
   this.readerMode.selectFontSize()
   ..etc
}
```

File: javascripts/discourse/components/reader-mode-options.gjs [hunk 2]
```diff
@@ -3,134 +3,147 @@ import { tracked } from "@glimmer/tracking";
...
+                    "true"
+                  }}
+                />
+              </div>
+              <div class="reader-mode-options__section-item">
+                <DButton
+                  @action={{this.readerMode.decrementOffset}}
+                  @preventFocus={{true}}
+                  @icon="compress-alt"
+                  @class="btn-flat reader-mode-options__item-button decrease-width text-changes"
```
Comment (janzenisaac) [inline javascripts/discourse/components/reader-mode-options.gjs:103]:
```suggestion
                  class="btn-flat reader-mode-options__item-button decrease-width text-changes"
```

File: javascripts/discourse/components/reader-mode-options.gjs [hunk 3]
```diff
@@ -3,134 +3,147 @@ import { tracked } from "@glimmer/tracking";
...
+                  @disabled={{if (eq this.readerMode.offsetIncrement 0) "true"}}
+                />
+                <DButton
+                  @action={{this.readerMode.incrementOffset}}
+                  @preventFocus={{true}}
+                  @icon="arrows-alt-h"
+                  @class="btn-flat reader-mode-options__item-button increase-width text-changes"
```
Comment (janzenisaac) [inline javascripts/discourse/components/reader-mode-options.gjs:110]:
```suggestion
                  class="btn-flat reader-mode-options__item-button increase-width text-changes"
```

File: javascripts/discourse/components/reader-mode-options.gjs [hunk 4]
```diff
@@ -3,134 +3,147 @@ import { tracked } from "@glimmer/tracking";
...
-                  min="1"
-                  max="208"
-                  step="1"
-                  value="0"
-                  {{on "input" this.selectOffset}}
+                <DButton
+                  @action={{this.readerMode.toggleLightMode}}
+                  @preventFocus={{true}}
+                  @icon="circle"
+                  @class={{concatClass
```
Comment (janzenisaac) [inline javascripts/discourse/components/reader-mode-options.gjs:125]:
```suggestion
                  class={{concatClass
```

File: javascripts/discourse/components/reader-mode-options.gjs [hunk 5]
```diff
@@ -3,134 +3,147 @@ import { tracked } from "@glimmer/tracking";
...
+                    "btn-flat reader-mode-options__item-button sepia-mode color-mode"
+                    (if (eq this.readerMode.colorMode "sepia") "active")
+                  }}
+                />
+                <DButton
+                  @action={{this.readerMode.toggleDarkMode}}
+                  @preventFocus={{true}}
+                  @icon="circle"
+                  @class={{concatClass
```
Comment (janzenisaac) [inline javascripts/discourse/components/reader-mode-options.gjs:143]:
```suggestion
                  class={{concatClass
```

File: javascripts/discourse/components/reader-mode-toggle.gjs [hunk 6]
```diff
@@ -1,40 +1,27 @@
...
+  @service readerMode;
 
   get bodyClassText() {
-    return this.isTransitioning
+    return this.readerMode.isTransitioning
       ? "reader-mode-transitioning reader-mode"
-      : this.readerModeActive
+      : this.readerMode.readerModeActive
       ? "reader-mode"
       : "";
```
Comment (janzenisaac) [inline javascripts/discourse/components/reader-mode-toggle.gjs:19]: IMO this reads a bit better

```js
if (this.readerMode.isTransitioning) {
  return "reader-mode-transitioning reader-mode";
} else if (this.readerMode.readerModeActive) {
  return "reader-mode";
} else {
  return "";
}
```

**Review by janzenisaac — COMMENTED** [review]
Love this feature

File: javascripts/discourse/components/reader-mode-options.gjs (hunk 1 above)
Comment (jordanvidrine) [reply javascripts/discourse/components/reader-mode-options.gjs:35]: Love that idea

**Review by janzenisaac — APPROVED** [review]

**Pull request merged**